### DISCUSSION TITLE
What features from default Prusa firmware we will lose?

### DISCUSSION THREAD

**Discussion by 0x414c49** [dz0ny/klipper-prusa-mk3s#33] [General]
I'm thinking to pull the trigger and Klipperise my MK3S+, I'm curios to know what feature I'll will lose.

**Comment by confuzled-dev** [comment]
Nearly all the features accessible through the menu.

However, depending on you Google fu, knowledge of python, or willingness to
learn python , the sky is you limit as to what can be built back.

I switched and won't be going back

**Comment by 0x414c49** [comment]
Thanks for the info. I'm talking about things like bed profile (like smooth1, sating1) or things like collision detection. I'm a software engineer so I'm not afraid of these.

Just wanted to know what I'll get and what I'll loose. Like why I should switch to Klipper? (except the input shaping and pressure advance).

**Comment by confuzled-dev** [comment]
Again, anything you do from the lcd screen, you basically lose.

This means storing and quickly changing bed profiles, calibration, etc,
however, these are not necessarily difficult things to implement.

Now, as to why you you want to install klipper, in it's lowest form, I look
at this like Android vs Apple; some people just want to control their
hardware. In addition to this, even if you only installed klipper and
didn't tune pressure advance or input shaping, you can still get better
prints. This is because 8bit boards, from what I understand, use a lot of
shortcuts it achieving high speed that the SBC doesn't need because of it's
increased horsepower.

I think the ideal use case and benefits for klipper on prusa are as follows:

Adaptive Mesh (only mesh where you are printing.
Increased print quality at about a 20-25% reduction in print time.
Same quality at lower (sometimes substantially) print times.
Increased control over hardware.


The prusa frame doesn't really work well with high print speeds so if you
are okay with increasing both quality and speed moderately, klipper is a
great option.

Just don't expect it to be a speed demon.

I am printing in about 50-60% of my old print times since installing
klipper. I have introduced a slight bit of ringing, but the prints would be
an absolute mess on stock firmware at these speeds
